pkg/types/types.go

```diff
@@ -39,6 +39,7 @@ const (
 	StateStarting         = State("starting")
 	StateInProgress       = State("in-progress")
 	StateFailed           = State("failed")
+	StateFailedAndCleanUp = State("failed-and-cleanup")
```

Comment:
no it is not used.
I added it because `file state` was always copied from `sync file status` (here) to `CR status` (longhorn-manager controller)
But `failed-and-cleanup` is the first state introduced in `CR status` only
I am not sure if we need to add this state here to make these two parts have same state list
WDYT?

Reply:
We should have the same state list for sure, but this kind of way cause a little confusion to have duplicated list in two places. The better way should have one place for these states then as a share lib to use in another dependents.

It's fine here for the current design.